src/freenet/clients/http/UserAlertsToadlet.java

```diff
@@ -47,7 +50,16 @@ public void handleMethodPOST(URI uri, HTTPRequest request, ToadletContext ctx) t
 			int userAlertHashCode = request.getIntPart("disable", -1);
 			ctx.getAlertManager().dismissAlert(userAlertHashCode);
 		}
-		headers.put("Location", ".");
+		
+		
+		String redirect;
+		try {
+			redirect = request.getPartAsStringThrowing("redirectToAfterDisable", 1024);
+		} catch (SizeLimitExceededException | NoSuchElementException e) {
```

Comment:
Why is a JNDI-specific exception throw here? Please use `IllegalArgumentException` if there’s anything wrong an argument.

Reply:
I will merge it for now and fix the API later

Reply:
@nextgens You don't have to fix my crap API, I've already filed a bug about it and assigned it to myself so I will deal with it once the current WoT build is done. Thanks for merging!